providers/src/airflow/providers/google/cloud/hooks/cloud_sql.py

```diff
@@ -1187,3 +1195,33 @@ def free_reserved_port(self) -> None:
         if self.reserved_tcp_socket:
             self.reserved_tcp_socket.close()
             self.reserved_tcp_socket = None
+
+    def _get_iam_db_login(self) -> str:
+        """Get an IAM login for Cloud SQL database."""
+        if not self.cloudsql_connection.login:
+            raise AirflowException("The login parameter needs to be set in connection")
+
+        if self.database_type == "postgres":
+            return self.cloudsql_connection.login.split(".gserviceaccount.com")[0]
+        else:
+            return self.cloudsql_connection.login.split("@")[0]
+
+    def _generate_login_token(self, service_account) -> str:
+        """Generate an IAM login token for Cloud SQL and return the token."""
+        cmd = ["gcloud", "sql", "generate-login-token", f"--impersonate-service-account={service_account}"]
```

Comment:
Do we use such approach with gcloud in other operators?
I wonder if we should specify that "gcloud" is a dependency for this operator (maybe in docs)?

I do not think that gcloud is dependency for apache-airflow-providers-google package, right?

Reply:
Yes we use this approach in our operators for example in [Dataflow](https://github.com/apache/airflow/blob/main/providers/src/airflow/providers/google/cloud/hooks/dataflow.py#L1240).

Yes you are right that `gcloud` is not a dependency for the `apache-airflow-providers-google` package.
I will add a warning in the docs that `gcloud` must be installed for using IAM auth.

Reply:
sg